thirdeye-spi/src/main/java/ai/startree/thirdeye/spi/authorization/AccessControl.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2022 StarTree Inc
+ *
+ * Licensed under the StarTree Community License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at http://www.startree.ai/legal/startree-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT * WARRANTIES OF ANY KIND,
+ * either express or implied.
+ * See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package ai.startree.thirdeye.spi.authorization;
+
+import java.net.http.HttpHeaders;
+
+public interface AccessControl {
+
+  boolean hasAccess(
+      AccessControlIdentifiers identifiers,
+      AccessType accessType,
+      HttpHeaders httpHeaders
```

Comment:
Instead of `httpHeaders`, can we structure the exact information that we need in the interface for the implementation to be able to decide on the access control